src/openrct2/management/Research.cpp

```diff
@@ -459,27 +459,27 @@ static void research_insert_unresearched(sint32 entryIndex, sint32 category)
             memmove(researchItem + 1, researchItem, (researchItem2 - researchItem + 1) * sizeof(rct_research_item));
 
             // Place new item
-            researchItem->rawValue = entryIndex;
-            researchItem->category   = category;
+            researchItem->rawValue = rawValue;
+            researchItem->category = category;
             break;
         }
     }
-    while (entryIndex != (researchItem++)->rawValue);
+    while (rawValue != (researchItem++)->rawValue);
 }
 
 /**
  *
  *  rct2: 0x00685826
  */
-static void research_insert_researched(sint32 entryIndex, sint32 category)
+static void research_insert_researched(sint32 rawValue, uint8 category)
 {
     rct_research_item * researchItem, * researchItem2;
 
     researchItem = gResearchItems;
     // First check to make sure that entry is not already accounted for
     for (; researchItem->rawValue != RESEARCHED_ITEMS_END; researchItem++)
     {
-        if (researchItem->rawValue == entryIndex)
+        if ((researchItem->rawValue & 0xFFFFFF) == (rawValue & 0xFFFFFF))
```

Comment:
Do we/can we have a define for this?

Reply:
This is all bound to be replaced this year.